shared/ProtoYaml/proto-yaml.h

```diff
@@ -0,0 +1,48 @@
+/** Copyright (C) 2019 Josh Ventura
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include <yaml-cpp/yaml.h>
+#include <google/protobuf/message.h>
+
+#include <algorithm>
+#include <iostream>
+#include <fstream>
+#include <type_traits>
+#include <map>
+#include <set>
+
+namespace egm {
```

Comment:
I'm not sure this is really part of the egm namespace as we reserved gmk:: & gmx:: for functions like loadproject() / writeproject(). There's no real harm in putting it under egm namespace but this is more general purpose.